modules/url-titles.js

```diff
@@ -0,0 +1,47 @@
+const cheerio = require('cheerio')
+const request = require('request')
+var getTitles = function (bot, to, nick, urls, titles) {
+  const maxSize = 1024
+  var size = 0
+  var response = ''
+  var req = request(urls[0])
+  // Check the size of the downloaded data regularly
+  req.on('data', chunk => {
+    size += chunk.length
+    response += chunk.toString()
+    // After maxSize bytes, abort
+    if (size > maxSize) {
+      req.abort()
+    }
+  }).on('error', () => {
+    // If there was an error, don't bother trying to get the title
+    // But still process the rest of the URLs
+    if (urls.length > 1) {
+      getTitles(bot, to, nick, urls.slice(1), titles)
```

Comment:
At the moment it will only do one lookup at a time. Would it be better to do it asynchronously?